### PR TITLE
Component containers

### PR DESCRIPTION
**Pull request by Tapiiri**
Split components to components and containers - components are dumb and render stuff, and containers figure out logic and state handling.

Some containers are a bit empty and unneeded right now, but they would be filled soon enough, so they could be let stick around imo.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-import "./App.css";
-import { string, shape } from "prop-types";
+import "../App.css";
+import { string } from "prop-types";
 
-import Table from "./components/Table";
-import UserForm from "./components/UserForm";
+import Sitsit from "../containers/Sitsit";
+import UserForm from "../containers/UserForm";
+import { userInfo } from "../propTypes";
```
Comment (VonLatvala) [inline src/components/App.js:7]: Hm, an odd convention.

File: src/components/Sitsit.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import "../App.css";
+
+import Table from "./Table";
+import { arrayOf, number } from "prop-types";
+import { table } from "../propTypes";
+
+const DEFAULT_NUMBER_OF_TABLES = 2;
```
Comment (VonLatvala) [inline src/components/Sitsit.js]: `src/constants/defaults.js` pls, and then take it as a prop?

File: src/components/Table.js [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+    seatLocations: array.isRequired,
+    numberOfSeats: number.isRequired,
+    tableEndSeatAllowed: bool.isRequired,
+};
 
 const DEFAULT_NUMBER_OF_SEATS = 10;
 
 function createSeatLocations(numberOfSeats, tableEndSeatAllowed = false) { // eslint-disable-line
-  // eslint-disable-line
+    // eslint-disable-line
```
Comment (VonLatvala) [inline src/components/Table.js]: no need for this line

File: src/components/UserForm.js [hunk 4]
```diff
@@ -1,34 +1,28 @@
 import React from "react";
-import { connect } from "react-redux";
 import "../App.css";
-import { submitUserInfoAction } from "../actions/submitUserInfoAction";
-import { updateUserInfoFormAction } from "../actions/updateUserInfoFormAction";
-import { func, shape, string } from "prop-types";
+import { userInfo, userFormInfo } from "../propTypes";
```
Comment (VonLatvala) [inline src/components/UserForm.js:3]: Hm okay now I see why these are in their own file. Splendid!

**Review by VonLatvala — CHANGES_REQUESTED** [review]
Minimal nags

File: src/components/Sitsit.js (hunk 2 above)
Comment (Tapiiri) [reply src/components/Sitsit.js]: Sure!

File: src/components/App.js (hunk 1 above)
Comment (Tapiiri) [reply src/components/App.js:7]: Which one?

File: src/containers/App.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { connect } from "react-redux";
+import App from "../components/App";
+
+const mapStateToProps = state => ({
+    seat: state.userReducer.seat || "",
```
Comment (VonLatvala) [inline src/containers/App.js:5]: Is this a good convention?

Comment (VonLatvala) [reply src/containers/App.js:5]: maybe put it into the reducer? idk

File: src/containers/UserForm.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { connect } from "react-redux";
+import { submitUserInfoAction } from '../actions/submitUserInfoAction';
+import { updateUserInfoFormAction } from '../actions/updateUserInfoFormAction';
+import UserForm from "../components/UserForm";
+
+
+const mapStateToProps = state => ({
+    info: state.userReducer.info || {},
```
Comment (VonLatvala) [inline src/containers/UserForm.js:7]: same here, check if ok if not fixxxxx

**Review by VonLatvala — APPROVED** [review]
Please check if the conventions are correct, if they are, go ahead and merge. LGTM

**Review by Earlo — APPROVED** [review]
dewit

**Pull request merged**